### DISCUSSION TITLE
Is there a better way to replace online media resources with local files (other than using a custom scheme)?

### DISCUSSION THREAD

**Discussion by RadarNyan** [cefsharp/CefSharp#4890] [Q&A] [answered]
I'm currently rewriting an old project using CefSharp.WinForms. It's a simple browser that specializes in one thing: replacing online media resources with local files.

My old implementation was:
1. Match URLs in ```OnBeforeResourceLoad``` and modifying the scheme part ```request.Url``` before returning ``` CefReturnValue.Continue```
2. Handle the request from ```ResourceHandler``` of my custom scheme, basically creating an HTTP 200 / 206 response.
This implementation results in a 307 redirect (as shown in developer tools) from the original URL (https://...) to my custom URL (myscheme://...)

The old code works, but I wonder if there's a better way to achieve this. For example, is it possible to return my forged response directly without having any redirects (all while not sending any traffic to the original URL)?

**Comment by amaitland** [comment, marked as answer]
https://github.com/cefsharp/CefSharp/wiki/General-Usage#request-interception
